pkg/kubelet/nodestatus/setters.go

```diff
@@ -456,6 +457,10 @@ func ReadyCondition(
 			Message:           "kubelet is posting ready status",
 			LastHeartbeatTime: currentTime,
 		}
+		if !isRuntimeInitialized() {
+			// do not report container runtime status until it is initialized.
+			return fmt.Errorf("ContainerRuntimeNotInitializedError: will try to update nodeready status later")
```

Comment:
How about just log error like this: 

```suggestion
			return fmt.Errorf("Container runtime is not initialized, will try to update nodeready status later")
```

Reply:
Done